### DISCUSSION TITLE
Transfer/replace content for deleted elements in required fields

### DISCUSSION THREAD

**Discussion by bymayo** [craftcms/cms#14627] [Ideas] [answered]
We've had an issue crop up a few times on clients sites where say we have a required element field, but if an element is deleted it removes it from the required field, making the entry invalid and sometimes breaking the whole site.

A good example is if we have a user field, for "Additional Authors" that is required, and an additional author (User) is added, but later down the line this user is deleted either by themselves, or another admin, this "Additional Authors" now is blank within the entry, but it is required so it needs a value to work.

Obviously we can get around it with a conditional in the templating, but in most cases we only do a required field if it's linked across the site in multiple places and NEEDS to be there.

One way around this would be to check if the element is linked to a required field, and if it is, ask the user deleting the element, to choose it's replacement, or some how saving it's state/content somewhere.

**Comment by bymayo** [comment, marked as answer]
@brandonkelly Does 5.10.0 "element deletion blockers" feature solve this issue?

**Reply by brandonkelly** [reply]
Yep! Sorry, forgot to update this.
